drivers/mbox/mbox_stm32_hsem.c

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2024 Celina Sophie Kalus <hello@celinakalus.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+
+#include "stm32_hsem.h"
+
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+LOG_MODULE_REGISTER(mbox_stm32_hsem_ipc);
+
+#define DT_DRV_COMPAT st_mbox_stm32_hsem
+
+#define HSEM_CPU1                   1
+#define HSEM_CPU2                   2
+
+#if DT_NODE_EXISTS(DT_NODELABEL(cpu0))
+#define HSEM_CPU_ID HSEM_CPU1
+#elif DT_NODE_EXISTS(DT_NODELABEL(cpu1))
+#define HSEM_CPU_ID HSEM_CPU2
+#else
+#error "Neither cpu0 nor cpu1 defined!"
+#endif
+
+#if HSEM_CPU_ID == HSEM_CPU1
+#define MBOX_TX_CHANNEL 0
+#define MBOX_TX_HSEM_ID CFG_HW_IPM_CPU2_SEMID
```

Comment:
This sounds like an awesome idea and I would love to hear the others opinions. If this is something you decide should be done, I would gladly give implementing it a shot.

Reply:
A good idea, but for consistency it would require to update other usages of CFG_FOO definitions provided [here](soc/st/stm32/common/stm32_hsem.h), and in those areas, interest is lower.

Reply:
I have implemented this now; Multiple channels are now thus possible. Please let me know what you think.

Reply:
Reverted after result from another thread. Closing this here.